app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx

```diff
@@ -0,0 +1,307 @@
+import * as COLORS from 'common/constants/colors';
+import * as d3 from 'd3-selection';
+import ReactDOMServer from 'react-dom/server';
+import { C3Chart } from '../common/c3chart';
+import { TooltipWrapper, TooltipPassingContent } from '../common/tooltip';
+import './passingRatePerLaunch.scss';
+import { Legend } from '../common/legend';
+
+@injectIntl
+@connect(
+  (state) => ({
+    project: activeProjectSelector(state),
+    getDefectLink: (params) => defectLinkSelector(state, params),
+    getStatisticsLink: (name) => statisticsLinkSelector(state, { statuses: [name] }),
+  }),
+  {
+    redirect,
+  },
+)
+export class PassingRatePerLaunch extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    redirect: PropTypes.func.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    height: PropTypes.number,
+    project: PropTypes.string.isRequired,
+    container: PropTypes.instanceOf(Element).isRequired,
+    observer: PropTypes.object.isRequired,
+    getDefectLink: PropTypes.func.isRequired,
+    getStatisticsLink: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    isPreview: false,
+    height: 0,
+  };
+
+  state = {
+    isConfigReady: false,
+  };
+
+  componentDidMount() {
+    this.props.observer.subscribe('widgetResized', this.resizeChart);
+    this.getConfig();
+  }
+
+  componentWillUnmount() {
+    this.node.removeEventListener('mousemove', this.getCoords);
+    this.props.observer.unsubscribe('widgetResized', this.resizeChart);
+  }
+
+  onChartCreated = (chart, element) => {
+    this.chart = chart;
+    this.node = element;
+
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      d3.selectAll('.c3-chart-texts text, .c3-chart-arc text').each(() => {
+        d3.select(this).remove();
+      });
+      return;
+    }
+
+    this.node.addEventListener('mousemove', this.getCoords);
+
+    if (!this.props.isPreview) {
+      this.resizeHelper(this.node);
+    }
+    // this.resizeChart();
+  };
+
+  onMouseOut = () => {
+    this.chart.revert();
+  };
+
+  onMouseOver = (id) => {
+    this.chart.focus(id);
+  };
+
+  getPosition = (d, width, height) => {
+    const rect = this.node.getBoundingClientRect();
+    const left = this.x - rect.left - width / 2;
+    const top = this.y - rect.top - height;
+
+    return {
+      top: top - 8,
+      left,
+    };
+  };
+
+  getProcessedData = (data, isPreview, colors) => {
+    const itemNames = ['Passed', 'Failed'];
+    const columns = [
+      ['Passed', parseInt(data.content.result.passed, 10)],
+      [
+        'Failed',
+        parseInt(data.content.result.total, 10) - parseInt(data.content.result.passed, 10),
+      ],
+    ];
+    const columnData = { Passed: columns[0][1], Failed: columns[1][1] };
+    const chartData = {
+      columns,
+      groups: [itemNames],
+      type: data.content_parameters.widgetOptions.viewMode === 'barMode' ? 'bar' : 'pie',
```

Comment:
`contentParameters`
WS is switching to camel case everywhere

Reply:
This content_parameters comes from the demodata. 
U suggesting to converting to it to camel case? Or the backend give us in the future camel case json?

Reply:
In the near future all responses will be in camel case.